From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sleigh-InSPECtor <sleighinspector@outlook.com>
Date: Mon, 15 Jul 2024 10:37:20 +0930
Subject: [PATCH] 6607: AArch32: addrmode3 incorrectly had load in address
 pointer export

aarch32: addrmode3 constructor with reloff should export a constant not a memory address
---
 Ghidra/Processors/ARM/data/languages/ARMinstructions.sinc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Processors/ARM/data/languages/ARMinstructions.sinc b/Ghidra/Processors/ARM/data/languages/ARMinstructions.sinc
index 057cbbdb9d..443f12a7aa 100644
--- a/Ghidra/Processors/ARM/data/languages/ARMinstructions.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARMinstructions.sinc
@@ -1784,13 +1784,13 @@ addrmode2: [rn],-addr2shift is I25=1 & P24=0 & U23=0 & W21=1 & rn & addr2shift	{
 addrmode3: [reloff]		is P24=1 & U23=1 & c2122=2 & c1619=15 & immedH & immedL
   [ reloff=inst_start+8+((immedH<<4) | immedL);]
 {
-  export *:4 reloff;
+  export *[const]:4 reloff;
 }
 
 addrmode3: [reloff]		is P24=1 & U23=0 & c2122=2 & c1619=15 & immedH & immedL
   [ reloff=inst_start+8-((immedH<<4) | immedL);]
 {
-  export *:4 reloff;
+  export *[const]:4 reloff;
 }
 
 addrmode3: [rn,"#"^off8]		is P24=1 & U23=1 & c2122=2 & rn & immedH & immedL
-- 
2.45.1

